lib/_emerge/actions.py

```diff
@@ -772,6 +772,89 @@ def action_config(settings, trees, myopts, myfiles):
     return retval
 
 
+def get_libc_version(vardb):
+    libcver = []
+    libclist = set()
+    for atom in expand_new_virt(vardb, portage.const.LIBC_PACKAGE_ATOM):
+        if not atom.blocker:
+            libclist.update(vardb.match(atom))
+    if libclist:
+        for cpv in sorted(libclist):
+            libc_split = portage.catpkgsplit(cpv)[1:]
+            if libc_split[-1] == "r0":
+                libc_split = libc_split[:-1]
+            libcver.append("-".join(libc_split))
+        return libcver
+    return ["unavailable"]
+
+
+def get_profilever(portdir, profile, vardb):
+    profilever = None
+    repositories = vardb.settings.repositories
+    if profile:
+        profilever = relative_profile_path(portdir, profile)
+        if profilever is None:
+            try:
+                for parent in portage.grabfile(os.path.join(profile, "parent")):
+                    profilever = relative_profile_path(
```

Comment:
We have >2 levels of nesting where, I think it would make it more readable to refactor the contents of this for loop into a function